### DISCUSSION TITLE
helm-s3-proxy: A HTTP compatible read proxy for the helm-s3 plugin.

### DISCUSSION THREAD

**Discussion by cep21** [fluxcd/flux2#2428] [Show and tell]
https://github.com/cresta/helm-s3-proxy

This is a project that allows you to keep your existing CI/CD system that uploads helm charts to s3 while also migrating to flux2.  Our current setup does not use --relative paths, so we need to do the extra step of processing the index.yaml file and modifying the urls to also point to the proxy.  We also cache the index.yaml content and use e-tag to check for updates.  The intended use is to install this inside the cluster and serve helm charts for flux2 running inside the cluster.

**Comment by kingdonb** [comment]
Thanks for making this and sharing it!

I wanted to ask how does this compare to the suggested solution in the HelmRelease guide, did you see this:

https://fluxcd.io/docs/guides/helmreleases/#cloud-storage

How does the approach suggested in this doc meet your needs or not, compared to the `helm-s3-proxy`?

(Does the proxy work with Flux work now, or are you suggesting there might be room for additional support from Flux to be able to use this proxy as a Source type?)

**Reply by cep21** [reply]
> How does the approach suggested in this doc meet your needs or not, compared to the helm-s3-proxy?

Two main problems I had:

1. I want a solution that works with my current CI/CD pipeline. Using helm-s3-proxy means I don't have to also upload charts to another repository or upload old existing charts
2. We have lots of repositories.  (one per chart type).  Using chart museum is awkward for this setup, requiring setups for each. `helm-s3-proxy` and helm's `s3` plugin work seamlessly for a large number of chart repositories.

**Reply by cep21** [reply]
> Does the proxy work with Flux work now

The `helm-s3-proxy` will just work with flux.  Here is the example flux2 resource.  Of special note: the URL is a directory in the S3 bucket (we have dozens of `HelmRepository` objects in our kubernetes cluster)

```
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: HelmRepository
metadata:
  name: chart2
spec:
  interval: 1m0s
  url: http://helm-s3-proxy.helm-s3-proxy.svc.cluster.local/chart2
```
